src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,83 @@
+package duke;
+
+import java.util.Arrays;
+
+/**
+ * Deals with making sense of the user command and converting input into a format understandable by the Duke program.
+ */
+public class Parser {
+    private static void fillNullElementsWithEmptyString(String[] arr) {
+        for (int i = 0; i < arr.length; i++) {
+            if (arr[i] == null) {
+                arr[i] = "";
+            }
+        }
+    }
+
+    /**
+     * Parses an input string into a format that Duke can understand (An array of String).
+     *
+     * @param input A user input string.
+     * @return A String array containing the parsed user input.
+     * @throws BannedDukeCharacterException If input contains a '|' character.
+     * @throws InvalidDukeInputException If input does not contain a recognised Duke command.
+     * @throws MissingDukeInputException If input contains a command that is lacking supporting information.
+     */
+    public static String[] parseInput(String input)
+            throws BannedDukeCharacterException,
+            InvalidDukeInputException,
+            MissingDukeInputException {
+        String[] parsedOutput = new String[6];
+        input = input.trim();
+        if (input.contains("|")) {
+            throw new BannedDukeCharacterException("|");
+        }
+        switch (input) {
+        case "bye":
+        case "list":
+            parsedOutput[0] = input;
+            return parsedOutput;
+        default:
+            String[] str = input.split(" ", 2);
+            String cmd = str[0];
+            if (str.length == 1
+                    && (cmd.equals("mark")
+                    || cmd.equals("unmark")
+                    || cmd.equals("delete")
+                    || cmd.equals("todo")
+                    || cmd.equals("deadline")
+                    || cmd.equals("event"))) {
+                throw new MissingDukeInputException(cmd);
+            }
+            parsedOutput[0] = cmd;
+
+            switch (cmd) {
```

Comment:
Perhaps parsing the command can be done prior to the first `switch` statement to avoid [deep nesting](https://nus-cs2103-ay2223s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-deep-nesting)?